moveit_planners/pilz_industrial_motion_planner/src/trajectory_blender_transition_window.cpp

```diff
@@ -85,7 +86,10 @@ bool pilz_industrial_motion_planner::TrajectoryBlenderTransitionWindow::blend(
   }
   trajectory_msgs::JointTrajectory blend_joint_trajectory;
   moveit_msgs::MoveItErrorCodes error_code;
-  if (!generateJointTrajectory(req.first_trajectory->getFirstWayPointPtr()->getRobotModel(),
+
+  const moveit::core::RobotModelConstPtr& robot_model = req.first_trajectory->getFirstWayPointPtr()->getRobotModel();
+  const planning_scene::PlanningSceneConstPtr scene(new planning_scene::PlanningScene(robot_model));
```

Comment:
Here a new (empty) scene is created from scratch. Isn't that exactly what you wanted to avoid?

Reply:
It is... If you have an idea of which route of function calls could be used to pass the scene down here that'd be super useful, otherwise I can have a dig through the code

Reply:
You are right, you probably have to pass it down quite along way. Nonetheless It's definitely required here or trajectory blending can yield invalid trajectories that were fine before.
I expect the `PlanComponentsBuilder` will always operate on a single PlanningScene (with varying robot states), so what about replacing `PlanComponentsBuilder::setRobotModel` by `PlanComponentsBuilder::setPlanningScene`?
I did not try it, but it seems reasonable at a glance.